compiler/luci-interpreter/src/kernels/FullyConnected.cpp

```diff
@@ -231,5 +249,32 @@ void FullyConnected::evalHybridWI4AF32() const
     getTensorShape(output()), getTensorData<float>(output()));
 }
 
+void FullyConnected::evalHybridWU4AF32() const
+{
+  float activation_min{};
+  float activation_max{};
+  calculateActivationRange(_params.activation, &activation_min, &activation_max);
+
+  tflite::FullyConnectedParams params{};
+  params.float_activation_min = activation_min;
+  params.float_activation_max = activation_max;
+  params.weights_format = tflite::FullyConnectedWeightsFormat::kDefault;
+
+  const auto *weights_uint4 = getTensorData<uint8_t>(weights());
+  auto *weights_float = getTensorData<float>(scratch());
+  const Shape &weights_shape = weights()->shape();
+  const auto wscale = weights()->scale();
+  const auto wzeropoint = weights()->zero_point();
```

Comment:
`scale()` and `zero_point()` is for per-tensor quantization. Per-channel quantization is not considered. You have to dequantize weights using `scales()` and `zero_points` according to `quantized_dimension`.

It seems that the same problem would also appear in int4 FC.